src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -1379,3 +1529,229 @@ void CustomTextLayout::_OrderRuns()
 }
 
 #pragma endregion
+
+// Routine Description:
+// - Calculates the box drawing scale/translate matrix values to fit a box glyph into the cell as perfectly as possible.
+// Arguments:
+// - format -
+// - face -
+// - effect -
+// - fontScale - (optional, 1.0f default), if the given font face is going to be scaled versus the format, we need to know so we can compensate for that.
+// Return Value:
+// - S_OK, GSL/WIL errors, DirectWrite errors, or math errors.
+[[nodiscard]] HRESULT STDMETHODCALLTYPE CustomTextLayout::s_CalculateBoxEffect(IDWriteTextFormat* format, IDWriteFontFace1* face, IBoxDrawingEffect** effect, float fontScale) noexcept

```

Comment:
nit: com out pointer should be the last arg

Reply:
I had it as 2nd to last because optional arguments have to be last, I thought, and I have the scale as optional.

Reply:
Fixed.